src/test/java/com/vinyl/DeleteProductFromCartTest.java

```diff
@@ -0,0 +1,128 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Cart;
+import com.vinyl.model.Product;
+import com.vinyl.model.ProductCart;
+import com.vinyl.model.User;
+import com.vinyl.repository.CartRepository;
+import com.vinyl.repository.ProductCartRepository;
+import com.vinyl.repository.TokenRepository;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.util.Lists;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+import java.time.LocalDate;
+
+public class DeleteProductFromCartTest extends BaseIntegration {
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    private ProductCartRepository productCartRepository;
+
+    @Autowired
+    private TokenRepository tokenRepository;
+
+    private Product product;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndCartExists() {
+        Cart cart = cartSetup();
```

Comment:
I would move the cart setup to the setUp method, theoretically and explicitly delete de cart for testWhenUserIsLoggedInAndCartDoesNotExist

Reply:
done